airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -183,6 +184,110 @@ def path(self, **kwargs):
+
+    For example:
+    Line items are part of each 'invoice' record, so use Invoices stream because
+    it allows bulk extraction:
+        0.1.28 and below - 1 request extracts line items for 1 invoice (+ pagination reqs)
+        0.1.29 and above - 1 request extracts line items for 100 invoices (+ pagination reqs)
+
+    if line items object has indication for next pages ('has_more' attr)
+    then use current stream to extract next pages. In major cases pagination requests
+    are not performed because line items are fully reported in 'invoice' record
+
+    Example for InvoiceLineItems and parent Invoice streams, record from Invoice stream:
+        {
+          "created": 1641038947,    <--- 'Invoice' record
+          "customer": "cus_HezytZRkaQJC8W",
+          "id": "in_1KD6OVIEn5WyEQxn9xuASHsD",    <---- value for 'parent_id' attribute
+          "object": "invoice",
+          "total": 0,
+          ...
+          "lines": {    <---- sub_items_attr
+            "data": [
+              {
+                "id": "il_1KD6OVIEn5WyEQxnm5bzJzuA",    <---- 'Invoice' line item record
+                "object": "line_item",
+                ...
+              },
+              {...}
+            ],
+            "has_more": false,    <---- next pages from 'InvoiceLineItemsPaginated' stream
+            "object": "list",
+            "total_count": 2,
+            "url": "/v1/invoices/in_1KD6OVIEn5WyEQxn9xuASHsD/lines"
+          }
+        }
+    """
+
+    filter: Optional[Mapping[str, Any]] = None
+    add_parent_id: bool = False
+
+    @property
+    @abstractmethod
+    def parent(self) -> StripeStream:
+        """
+        :return: parent stream which contains needed records in <sub_items_attr>
+        """
+
+    @property
+    @abstractmethod
+    def parent_id(self) -> str:
+        """
+        :return: string with attribute name
+        """
+
+    @property
+    @abstractmethod
+    def sub_items_attr(self) -> str:
+        """
+        :return: string if single primary key, list of strings if composite primary key, list of list of strings if composite primary key consisting of nested fields.
+          If the stream has no primary keys, return None.
+        """
+
+    def request_params(self, stream_slice: Mapping[str, Any] = None, **kwargs):
+        params = super().request_params(stream_slice=stream_slice, **kwargs)
+
+        # add 'starting_after' param
+        if not params.get("starting_after") and stream_slice and stream_slice.get("starting_after"):
+            params["starting_after"] = stream_slice["starting_after"]
+
+        return params
+
+    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+
+        parent_stream = self.parent(authenticator=self.authenticator, account_id=self.account_id, start_date=self.start_date)
+        for record in parent_stream.read_records(sync_mode=SyncMode.full_refresh):
```

Comment:
Can you help me understand what is going on, how come we're mentioning (forcing?) full_refresh modes here?

Is that because we are in the substream which can't be incremental? (only the parent level can?)